app/src/main/java/com/github/fribourgsdp/radio/GameActivity.kt

```diff
@@ -153,12 +154,13 @@ open class GameActivity : AppCompatActivity(), GameView, User.Loader {
     }
 
     open protected fun initVoiceChat(gameUid: Long) {
-        voiceChannel = VoiceIpEngineDecorator(this, MyIRtcEngineEventHandler(this))
-        val userId = Random.nextInt(100000000)
+
+        val map = mapIdToName.mapKeys { it.hashCode().absoluteValue }
+        if (!this::voiceChannel.isInitialized) voiceChannel = VoiceIpEngineDecorator(this, MyIRtcEngineEventHandler(this, map))
+        val userId = user.name.hashCode().absoluteValue
```

Comment:
Is that enough to guarantee uniqueness ?

Reply:
Good point. Yes, that's enough.